File: semgrep-rules/javascript/cors/testdata/cors_origin_reflection.js
const express = require("express");

// ruleid: javascript-cors-origin-reflection
function reflectOriginInline(req, res) {
    res.set("Access-Control-Allow-Origin", req.headers.origin);
}

// ruleid: javascript-cors-origin-reflection
function reflectOriginVariable(req, res) {
    const origin = req.headers.origin;
    res.set("Access-Control-Allow-Origin", origin);
}

// ok: javascript-cors-origin-reflection
function staticOrigin(req, res) {
    res.set("Access-Control-Allow-Origin", "https://trusted.example.com");
}

// ok: javascript-cors-origin-reflection
function wildcardOrigin(req, res) {
    res.set("Access-Control-Allow-Origin", "*");
}

File: semgrep-rules/python/cors/testdata/cors_origin_reflection.py
from django.http import HttpResponse


# ruleid: python-cors-origin-reflection
def reflect_origin_inline(request):
    response = HttpResponse()
    response["Access-Control-Allow-Origin"] = request.META["HTTP_ORIGIN"]
    return response


# ruleid: python-cors-origin-reflection
def reflect_origin_variable(request):
    origin = request.META["HTTP_ORIGIN"]
    response = HttpResponse()
    response["Access-Control-Allow-Origin"] = origin
    return response


# ok: python-cors-origin-reflection
def static_origin(request):
    response = HttpResponse()
    response["Access-Control-Allow-Origin"] = "https://trusted.example.com"
    return response


# ok: python-cors-origin-reflection
def wildcard_origin(request):
    response = HttpResponse()
    response["Access-Control-Allow-Origin"] = "*"
    return response
